src/sentry/static/sentry/app/views/settings/account/accountSecurity/accountSecurityEnroll.jsx

```diff
@@ -122,11 +122,17 @@ class AccountSecurityEnroll extends AsyncView {
   }
 
   getEndpoints() {
-    return [['authenticator', `${ENDPOINT}${this.props.params.authId}/enroll/`]];
+    return [];
```

Comment:
What you have here works, but I think it's possible to have done this within the constructs of `AsyncComponent`. You can define a function that checks if an error is ok: https://github.com/getsentry/sentry/blob/master/src/sentry/static/sentry/app/components/asyncComponent.jsx#L164

This would save us an extra request. Am I overlooking anything? Does this sound reasonable?

Reply:
Sounds good! Maybe not relevant for this case, but would there be a way in `AsyncComponent` to catch multiple errors or messages and act on them differently. If 'already enrolled' 400 redirect here, but if another error do something else.

Reply:
`allowError` gets passed the `error` object, so you should be able handle multiple errors no?

Reply:
...chatted offline and decided on adding a side effect to `allowError`